src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java

```diff
@@ -103,13 +104,19 @@ private static String sanitizeParam(String param) {
                 .subscribeOn(Schedulers.boundedElastic());
     }
 
-    private Mono<Network> getNetwork(UUID networkUuid, List<UUID> otherNetworkUuids) {
+    private Mono<Network> getNetwork(UUID networkUuid, List<UUID> otherNetworkUuids, UUID resultUuid) {
         Mono<Network> network = getNetwork(networkUuid);
         if (otherNetworkUuids.isEmpty()) {
             return network;
         } else {
             Mono<List<Network>> otherNetworks = Flux.fromIterable(otherNetworkUuids)
-                    .flatMap(this::getNetwork)
+                    .flatMap(uuid -> {
+                        if (resultUuid != null && cancelComputationRequests.get(resultUuid) != null) {
```

Comment:
actually all these are done in parallel so can you remove this check which is not useful ? sorry for asking to add it

Reply:
Done